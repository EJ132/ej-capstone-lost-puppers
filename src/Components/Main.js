import React, { Component } from 'react'
import '../Stylesheets/Main.css'
import Navbar from './Navbar'
import dogBone from '../Images/dogBone.png'
import community from '../Images/community.png'
import map from '../Images/map.png'

export default class Main extends Component {
    render() {
        return (
            <div>

                <Navbar />

                <section className="banner">
                    <div className="Overlay">
                        <h2>Lost Puppers</h2>
                        <p>Find your lost puppy today...</p>
                        <button>Get started</button>
                    </div>
                </section>

                <section className="SecondLayer">
                    <div className="three_Icons">
                        <div>
                            <img alt="paw" src={dogBone}></img>
                            <p>Lost your puppy, you came to the right place. 
                            Get started by clicking on the bone!</p>
                        </div>
                        <div>
                            <img alt="community building" src={community}></img>
                            <p>Dont have a puppy or maybe found one, click the 
                            community icon to find out more.</p>
                        </div>
                        <div>
                            <img alt="map" src={map}></img>
                            <p>Check the map to see if any puppies are wondering near you.</p>
                        </div>
                    </div>
                </section>

                <section className="ThirdLayer">
                    <div className="Third_Overlay">
                        <h2>Why use Lost Puppers?</h2>
                        <p>Lost Puppers is the new way to help you locate your pup that may have gotten loose.
                        And if you don't have a dog maybe you can help someone else out by finding their dog and 
                        earning a little bit of cash for yourself. Our pups are like our family and we will do anything
                        to find them and get them back home safe. So why not work as a community and use Lost Puppers!</p>
                        <div className="Main_lostDogs Row1">
                            <div className="lostDogCard">
                                <img alt="puppy" src="https://images.unsplash.com/photo-1523626797181-8c5ae80d40c2?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"></img>
                            </div>
                            <div className="lostDogCard mobile">
                                <img alt="puppy" src="https://images.unsplash.com/photo-1491604612772-6853927639ef?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"></img>
                            </div>
                        </div>
                        <div className="Main_lostDogs Row2">
                            <div className="lostDogCard mobile">
                                <img alt="puppy" src="https://images.unsplash.com/photo-1504826260979-242151ee45b7?ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60"></img>
                            </div>
                            <div className="lostDogCard">
                                <img alt="puppy" src="https://images.unsplash.com/photo-1558788353-f76d92427f16?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"></img>
                            </div>
                        </div>
                    </div>
                </section>

                <footer>&#169; EJ Gonzalez 2019</footer>
            </div>
        )
    }
}

// line 39...44 will be filled with dogCards (lost dog cards) components