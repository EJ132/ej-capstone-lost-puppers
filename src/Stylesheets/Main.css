/* FIRST SECTION DESIGN */

.banner {
    background: 
    url("../Images/homebackground.jpg") 
    no-repeat
    fixed;
    background-size: 2250px 1500px;
    position: relative;
    width: 100%;
    height: 95vh;
    padding: 20px;
    color: white;
    overflow: hidden;
}

.Overlay {
    padding: 20px;
    background-color: rgba(0, 0, 0, 0.7);
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
}

.banner h2 {
    margin-top: 200px;
    margin-left: 100px;
    font-size: 7rem;
}

.banner p {
    font-size: 2rem;
    margin-left: 150px;
}

.banner button {
    width: 200px;
    height: 75px;
    background-color: rgb(187, 177, 235);
    font-size: 1.2rem;
    font-weight: 600;
    border-radius: 10px;
    transition: all 0.9s;
    border: none;
    margin-top: 25px;
    margin-left: 250px;
}

.banner button:hover {
    cursor: pointer;
    background-color: #907DE8; 
    color: white;
}


/* SECOND SECTION DESIGN */

.SecondLayer {
    height: auto;
    color: white;
    padding: 20px;
    overflow: hidden;
}

.three_Icons {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    text-align: center;
    padding: 0px 20px 50px 20px;
}

.three_Icons img {
    padding: 10px;
    width: 210px;
    height: 210px;
}

.three_Icons p {
    font-size: 1.5rem;
    padding: 20px 40px 0px 40px;
}

/* THIRD SECTION DESIGN */

.ThirdLayer {
    display: flex;
    background: 
    url("../Images/homebackground2.jpg") 
    no-repeat
    center
    center
    fixed;
    background-size: cover;
    position: relative;
    width: 100%;
    height: 800px;
    padding: 20px;
}

.ThirdLayer h2 {
    margin-top: 175px;
    font-size: 4rem;
}

.ThirdLayer p {
    margin-top: 50px;
    margin-left: 75px;
    font-size: 1.2rem;
    width: 500px;
    text-align: center;
    word-spacing: 1px;
    line-height: 28px;
}

.Third_Overlay {
    padding: 20px;
    background-color: rgba(0, 0, 0, 0.7);
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    color: white;
}

.Main_lostDogs {
    margin-top: 100px;
    width: 45%;
    height: 300px;
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
}

.Row1 {
    position: absolute;
    top: 0;
    right: 20px;
    margin-top: 100px;
}

.Row2 {
    position: absolute;
    top: 350px;
    right: 20px;
    margin-top: 100px;
}

.Main_lostDogs div {
    background-color: #907DE8;
    width: 200px;
    height: 200px;
    border-radius: 50%;
    overflow: hidden;
}

.lostDogCard img{
    height: 100%;
    width: 100%;
    object-fit: fill;
}


/* FOOTER */

footer {
    width: 100%;
    color: white;
    text-align: center;
    height: 25px;
}

@media screen and (max-width: 768px) {
    /* FIRST SECTION DESIGN */

/* FIRST SECTION DESIGN */

.banner {
    background: 
    url("../Images/homebackground.jpg") 
    no-repeat
    center
    center
    fixed;
    background-size: 1500px 1000px;
    height: 95vh;
}

.Overlay {
    display: flex;
    flex-direction: column;
    justify-content: center;
    text-align: center;
}

.banner h2 {
    font-size: 3.3rem;
    margin-top: -125px;
    margin-left: 0;
}

.banner p {
    font-size: 1.2rem;
    margin-top: 10px;
    margin-left: 0;
}

.banner button {
    width: 175px;
    height: 60px;
    margin-right: auto;
    margin-left: auto;
    margin-top: 15px;
}


/* SECOND SECTION DESIGN */

.SecondLayer {
    height: auto;
    color: white;
    padding: 20px;
}

.three_Icons {
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    text-align: center;
    padding: 0px 20px 50px 20px;
}

.three_Icons img {
    padding: 10px;
    width: 210px;
    height: 210px;
}

.three_Icons p {
    font-size: 1rem;
    padding: 10px;
}

.three_Icons div {
    margin-top: 50px;
}

/* THIRD SECTION DESIGN */

.ThirdLayer {
    height: 825px;
}

.ThirdLayer h2 {
    margin-top: 40px;
    font-size: 1.9rem;
    text-align: center;
}

.ThirdLayer p {
    margin-top: 10px;
    margin-left: 0px;
    font-size: 1rem;
    width: 100%;
}

.Main_lostDogs {
    margin-top: 100px;
    width: 100%;
    height: 200px;
    display: flex;
    flex-direction: column;
}

.Row1 {
    position: unset;
    margin-top: 10px;
}

.Row2 {
    position: unset;
    margin-top: 10px;
}

.Main_lostDogs div {
    background-color: #907DE8;
    width: 150px;
    height: 150px;
    border-radius: 50%;
    overflow: hidden;
    margin-left: auto;
    margin-right: auto;
    margin-top: 0px;
}

.mobile {
    display: none;
}

.lostDogCard img{
    height: 100%;
    width: 100%;
    object-fit: fill;
}


/* FOOTER */

footer {
    width: 100%;
    color: white;
    text-align: center;
    height: 25px;
}
}

/* JUST FOR IPAD */

@media screen and (min-width: 769px) and (max-width: 1240px) {
    .banner h2 {
        margin-top: 250px;
    }

    .ThirdLayer h2 {
        width: 50%;
        font-size: 2rem;
        text-align: center;
    }

    .ThirdLayer p {
        margin-left: 5%;
        width: 40%;
        margin-top: 20px;
    }

    .Main_lostDogs {
        margin-top: 175px;
        height: 200px;
    }

    .Row2 {
        margin-top: 25px;
    }

    .Main_lostDogs div {
        width: 150px;
        height: 150px;
    }
}