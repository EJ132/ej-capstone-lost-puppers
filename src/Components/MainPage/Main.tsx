import React, { Component } from "react";
import "./Main.css";
import dogBone from "../../Images/dogBone.png";
import community from "../../Images/community.png";
import map from "../../Images/map.png";
import { Navigate, Link } from "react-router-dom";
import NavBar from "../NavBar/Navbar";
import TokenService from "../../services/token-service";

export default class Main extends Component {
  state = {
    redirect: false,
    where: "",
  };

  //redirects the page clicked on
  switchPage = (link) => {
    if (link === "login" && TokenService.hasAuthToken()) {
      link = "find";
    }

    this.setState({
      redirect: !this.state.redirect,
      where: link,
    });

    return;
  };

  render() {
    const { redirect, where } = this.state;

    //redirects the page to login
    if (redirect) {
      return <Navigate to={`/${where}`} />;
    }

    return (
      <div>
        <NavBar />

        <section className="banner">
          <div className="Overlay">
            <h2>Lost Puppers</h2>
            <p>Find your lost puppy today...</p>
            <button onClick={() => this.switchPage("login")}>
              Get started
            </button>
          </div>
        </section>

        <section className="SecondLayer">
          <div className="three_Icons">
            <div>
              <Link to="/login">
                <img alt="paw" src={dogBone}></img>
              </Link>
              <p>
                Lost your puppy, you came to the right place. Get started by
                clicking on the bone!
              </p>
            </div>
            <div>
              <Link to="/aboutus">
                <img alt="community building" src={community}></img>
              </Link>
              <p>
                Interest in how this site came to life or just want to explore
                more about the creator. Click on the racon above!
              </p>
            </div>
            <div>
              <Link to="/find">
                <img alt="map" src={map}></img>
              </Link>
              <p>Check the map to see if any puppies are wondering near you.</p>
            </div>
          </div>
        </section>

        <section className="ThirdLayer">
          <div className="Third_Overlay">
            <h2>Why use Lost Puppers?</h2>
            <p>
              Lost Puppers is the new way to help you locate your pup that may
              have gotten loose. And if you don't have a dog maybe you can help
              someone else out by finding their dog and earning a little bit of
              cash for yourself. Our pups are like our family and we will do
              anything to find them and get them back home safe. So why not work
              as a community and use Lost Puppers!
            </p>
            <div className="Main_lostDogs Row1">
              <div className="lostDogCard">
                <img
                  alt="puppy"
                  src="https://images.unsplash.com/photo-1523626797181-8c5ae80d40c2?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
                ></img>
              </div>
              <div className="lostDogCard mobile">
                <img
                  alt="puppy"
                  src="https://images.unsplash.com/photo-1491604612772-6853927639ef?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
                ></img>
              </div>
            </div>
            <div className="Main_lostDogs Row2">
              <div className="lostDogCard mobile">
                <img
                  alt="puppy"
                  src="https://images.unsplash.com/photo-1504826260979-242151ee45b7?ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60"
                ></img>
              </div>
              <div className="lostDogCard">
                <img
                  alt="puppy"
                  src="https://images.unsplash.com/photo-1558788353-f76d92427f16?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
                ></img>
              </div>
            </div>
          </div>
        </section>

        <footer>&#169; EJ Gonzalez 2019</footer>
      </div>
    );
  }
}
