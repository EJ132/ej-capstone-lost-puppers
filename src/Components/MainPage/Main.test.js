import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Main from './Main'

describe(`Main component`, () => {

  it('renders the home page by default', () => {
    const wrapper = shallow(<Main />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})
