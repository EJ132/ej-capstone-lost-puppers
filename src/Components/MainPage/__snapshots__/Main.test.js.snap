// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders the home page by default 1`] = `
<div>
  <Navbar />
  <section
    className="banner"
  >
    <div
      className="Overlay"
    >
      <h2>
        Lost Puppers
      </h2>
      <p>
        Find your lost puppy today...
      </p>
      <button
        onClick={[Function]}
      >
        Get started
      </button>
    </div>
  </section>
  <section
    className="SecondLayer"
  >
    <div
      className="three_Icons"
    >
      <div>
        <Link
          to="/login"
        >
          <img
            alt="paw"
            src="dogBone.png"
          />
        </Link>
        <p>
          Lost your puppy, you came to the right place. Get started by clicking on the bone!
        </p>
      </div>
      <div>
        <Link
          to="/aboutus"
        >
          <img
            alt="community building"
            src="community.png"
          />
        </Link>
        <p>
          Interest in how this site came to life or just want to explore more about the creator. Click on the racon above!
        </p>
      </div>
      <div>
        <Link
          to="/find"
        >
          <img
            alt="map"
            src="map.png"
          />
        </Link>
        <p>
          Check the map to see if any puppies are wondering near you.
        </p>
      </div>
    </div>
  </section>
  <section
    className="ThirdLayer"
  >
    <div
      className="Third_Overlay"
    >
      <h2>
        Why use Lost Puppers?
      </h2>
      <p>
        Lost Puppers is the new way to help you locate your pup that may have gotten loose. And if you don't have a dog maybe you can help someone else out by finding their dog and earning a little bit of cash for yourself. Our pups are like our family and we will do anything to find them and get them back home safe. So why not work as a community and use Lost Puppers!
      </p>
      <div
        className="Main_lostDogs Row1"
      >
        <div
          className="lostDogCard"
        >
          <img
            alt="puppy"
            src="https://images.unsplash.com/photo-1523626797181-8c5ae80d40c2?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
          />
        </div>
        <div
          className="lostDogCard mobile"
        >
          <img
            alt="puppy"
            src="https://images.unsplash.com/photo-1491604612772-6853927639ef?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
          />
        </div>
      </div>
      <div
        className="Main_lostDogs Row2"
      >
        <div
          className="lostDogCard mobile"
        >
          <img
            alt="puppy"
            src="https://images.unsplash.com/photo-1504826260979-242151ee45b7?ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60"
          />
        </div>
        <div
          className="lostDogCard"
        >
          <img
            alt="puppy"
            src="https://images.unsplash.com/photo-1558788353-f76d92427f16?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
          />
        </div>
      </div>
    </div>
  </section>
  <footer>
    Â© EJ Gonzalez 2019
  </footer>
</div>
`;
